src/partition/repart.c

```diff
@@ -72,6 +72,17 @@
 /* LUKS2 takes off 16M of the partition size with its metadata by default */
 #define LUKS2_METADATA_SIZE (16*1024*1024)
 
+#if !HAVE_LIBCRYPTSETUP
+        struct crypt_device {
+
+        };
```

Comment:
it should be sufficient to declare this without body, i.e.

```c
struct crypt_device;
```

Also, please move to first column, it's not customary to indent actual C code due to preprocessor `#ifdeffery` in our codebase